state/tx/indexer/batch.go

```diff
@@ -0,0 +1,32 @@
+package indexer
+
+import "github.com/tendermint/tendermint/types"
+
+// A Batch groups together multiple Index operations you would like performed
+// at the same time. The Batch structure is NOT thread-safe. You should only
+// perform operations on a batch from a single thread at a time. Once batch
+// execution has started, you may not modify it.
+type Batch struct {
+	Ops map[string]types.TxResult
```

Comment:
I think we should just replace this with a slice... it'll be faster and more deterministic.  We shouldn't get duplicate txs in the same batch anyways.